dev-infra/pr/merge/defaults/branches.ts

```diff
@@ -0,0 +1,201 @@
+      continue;
+    }
+    // Convert the version-branch into a SemVer version that can be used
+    // with the SemVer utilities. i.e. to determine semantic order.
+    const semver = parse(name.replace(releaseTrainBranchNameRegex, '$1.$2.0'));
+    // Collect all version-branches that should be considered for the latest
+    // version-branch (i.e. patch branch) or the feature-freeze/release-candidate.
+    // Branches that represent a minor more recent than the one in `next` are ignored.
+    if (semver !== null && majorVersionsToConsider.includes(semver.major)) {
+      branches.push({name, semver});
+    }
+  }
+
+  // Sort captured version-branches in descending order.
+  branches.sort((a, b) => rcompare(a.semver, b.semver));
+
+  let latestVersionBranch: string|null = null;
+  let releaseCandidateBranch: string|null = null;
+
+  // Iterate through the captured branches and find the latest non-prerelease branch and a
+  // potential release candidate branch. From the collected branches we iterate descending
+  // order (most recent semantic version-branch first). The first branch is either the latest
+  // active version branch (i.e. patch) or a feature-freeze/release-candidate branch. A FF/RC
+  // branch cannot be older than the latest active version-branch, so we stop iterating once
+  // we found such a branch. Otherwise, if we found a FF/RC branch, we continue looking for the
+  // next version-branch as that one is supposed to be the latest active version-branch. If it
+  // is not, then an error will be thrown due to two FF/RC branches existing at the same time.
+  for (const {name} of branches) {
+    const version = await getVersionOfBranch(repo, name);
+    const isPrerelease = version.prerelease[0] === 'rc' || version.prerelease[0] === 'next';
+    if (isPrerelease) {
+      if (releaseCandidateBranch !== null) {
+        throw Error(
+            `Unable to determine latest release-train. Found two consecutive ` +
+            `branches in feature-freeze/release-candidate phase. Expected "${name}" to not ` +
+            `be in feature-freeze/release-candidate mode.`);
+      } else if (version.major !== expectedReleaseCandidateMajor) {
+        throw Error(
+            `Discovered unexpected old feature-freeze/release-candidate branch. Expected no ` +
+            `version-branch in feature-freeze/release-candidate mode for v${version.major}`);
+      }
+      releaseCandidateBranch = name;
+    } else {
+      latestVersionBranch = name;
+      break;
+    }
+  }
+
+  if (latestVersionBranch === null) {
+    throw Error(
+        `Unable to determine the latest release-train. The following branches ` +
+        `have been considered: [${branches.join(', ')}]`);
+  }
+
+  return {releaseCandidateBranch, latestVersionBranch};
+}
+
+/** Gets the version of a given branch by reading the `package.json` upstream. */
+export async function getVersionOfBranch(repo: GithubRepo, branchName: string): Promise<SemVer> {
```

Comment:
```suggestion
export async function getBranchVersion(repo: GithubRepo, branchName: string): Promise<SemVer> {
```
you could probably shorten this a tiny bit

Reply:
I'd prefer leaving this as is unless you feel strongly about it. I think `branch-version` is less explicit than `version of branch`.